backend/cfg/merge_straightline_blocks.ml

```diff
@@ -77,6 +77,7 @@ let rec merge_blocks (removed : Label.Set.t)
             b1_block.terminator <- b2_block.terminator;
             b1_block.exn <- b2_block.exn;
             b1_block.can_raise <- b2_block.can_raise;
+            Cfg_with_layout.merge_sections cfg_with_layout b1_label b2_label;
```

Comment:
It is not immediately clear to me why
we want to merge the sections rather
than not apply the transformation if
the blocks are in different sections.

Reply:
No longer relevant since the cold attribute is on blocks.